### PR TITLE
Feature/polyfill

### PR DESCRIPTION
**Pull request by qwtel**
Opening PR so I can comment some lines

### CODE REVIEW

File: src/create-navigation-transition.ts [hunk 1]
```diff
@@ -36,13 +37,17 @@ export const NavigationCanIntercept = Symbol.for(
 export const NavigationUserInitiated = Symbol.for(
   "@virtualstate/navigation/userInitiated"
 );
+export const NavigationOriginalEvent = Symbol.for(
+  "@qwtel/navigation/originalEvent"
```
Comment (qwtel) [inline src/create-navigation-transition.ts:41]: No need to use my handle in your own repo :)

Comment (qwtel) [inline src/create-navigation-transition.ts:41]:
```suggestion
  "@virtualstate/navigation/originalEvent"
```

File: src/get-polyfill.ts [hunk 2]
```diff
@@ -0,0 +1,666 @@
...
+              return descriptor.get.call(this)
+            }
+          }
+      );
+      Object.defineProperty(
+          popStateEventPrototype,
+          "originalState", {
+            ...descriptor
+          }
+      );
```
Comment (qwtel) [inline src/get-polyfill.ts:429]: makes sense

File: src/get-polyfill.ts [hunk 3]
```diff
@@ -0,0 +1,666 @@
...
+    //   navigation.addEventListener(
+    //       "navigate",
+    //       // Add usage of intercept for initial navigation to prevent network navigation
+    //       (event) => event.intercept(Promise.resolve()),
+    //       { once: true }
+    //   );
+    //   await navigation.navigate(window.location?.href ?? "/", {
+    //     state: historyInitialState
+    //   }).finished
+    // }
```
Comment (qwtel) [inline src/get-polyfill.ts:497]: Very good!

File: src/get-polyfill.ts [hunk 4]
```diff
@@ -0,0 +1,666 @@
...
+          if (ignorePopState.delete(key)) return;
+          ignoreCurrentEntryChange.add(key);
+          let committed;
+          try {
+            committed = navigation.traverseTo(key).committed;
+          } catch (error) {
+            if (error instanceof InvalidStateError && !PERSIST_ENTRIES) {
+              // ignore the error
+              return;
+            }
```
Comment (qwtel) [inline src/get-polyfill.ts:609]: Btw, the the reason for ignoring this error is that this is common when not persisting entries.

Comment (qwtel) [inline src/get-polyfill.ts:609]: Btw, the the reason for ignoring this error is that this is common when not persisting entries.

File: src/create-navigation-transition.ts (hunk 1 above)
Comment (fabiancook) [reply src/create-navigation-transition.ts:41]: Cheers!

**Comment by fabiancook** [comment]
Thanks, closing this with the comments given

**Pull request closed without merging**